core-api/src/main/java/com/optimizely/ab/config/parser/AudienceGsonDeserializer.java

```diff
@@ -46,7 +46,7 @@ public Audience deserialize(JsonElement json, Type typeOfT, JsonDeserializationC
         String name = jsonObject.get("name").getAsString();
 
         JsonElement conditionsElement = jsonObject.get("conditions");
-        if (!conditionsElement.isJsonArray()) {
+        if (!typeOfT.toString().contains("TypedAudience")) {
             conditionsElement = parser.parse(jsonObject.get("conditions").getAsString());
```

Comment:
I think `typeOfT == null` allows `typedAudience.conditions` to be stringified and not just raw. I preferred the conditional statement without this bit!

Also, it might be cleaner to structure the above three lines using if-else, or to make the second assignment use the value from the first. It's not immediately obviously that the second assignment is supposed to (unconditionally, IMO) override the first in the case of typed audiences.

```
JsonElement conditionsElement;
if (!typeOfT.toString().contains("TypedAudience")) {
  conditionsElement = jsonObject.get("conditions");
} else {
  conditionsElement = parser.parse(jsonObject.get("conditions").getAsString());
}
```